tests integ: VLAN over a slave of a Bond

Nandan307 opened the pull request:
Signed-off-by: Nandan Kulkarni nakulkar@redhat.com

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

Nandan307 commented:
Somehow t`ests/integration/nm/linux_bridge_test.py::test_create_and_remove_bridge FAILED` is failing that too at  93%, no idea why....

EdDev commented on tests/integration/bond_test.py:
```diff
@@ -369,3 +369,11 @@ def test_bond_with_empty_ipv6_static_address(eth1_up):
     assertlib.assert_absent('bond99')
+
+
+def test_create_vlan_over_bond99_slave_interface(bond99_with_slave):
```
No need to mention the `slave` in the test name.

How about: `test_create_vlan_over_a_bond`

EdDev commented on tests/integration/bond_test.py:
```diff
@@ -369,3 +369,11 @@ def test_bond_with_empty_ipv6_static_address(eth1_up):
+def test_create_vlan_over_bond99_slave_interface(bond99_with_slave):
+    vlan_base_iface = BOND99
+    vlan_id = 101
+    port_name = '{}.{}'.format(vlan_base_iface, vlan_id)
```
`port` is bridge related. Use `vlan_iface_name`.

cathay4t commented:
@EdDev 

Is that legal to create VLAN over the slave of bond?

I was expecting `bond99.102` instead of `eth2.102` where `eth2` is slave of bond99.

cathay4t commented:
@Nandan307 Please click `Resolve conversation` if you believe you have fixed the issue in that conversation.

When you uploaded a new patch, please also leave a comment indication so with changes since last one. Example: https://github.com/nmstate/nmstate/pull/478#issuecomment-521677727

EdDev commented:
> Is that legal to create VLAN over the slave of bond?

Yes, it emphasizes that vlan/s and their base have existence and data flow relationship, but not a master/slave one. 

> I was expecting bond99.102 instead of eth2.102 where eth2 is slave of bond99.

The PR subject is correct, corresponding to the [issue](https://nmstate.atlassian.net/browse/NMSTATE-202). I think the confusion is due to the test name.

EdDev commented on tests/integration/bond_test.py:
```diff
@@ -369,3 +369,13 @@ def test_bond_with_empty_ipv6_static_address(eth1_up):
     assertlib.assert_absent('bond99')
+
+
+def test_create_vlan_over_a_bond(bond99_with_slave):
```
The name is incorrect, you create a VLAN over a bond slave.

EdDev commented on tests/integration/bond_test.py:
```diff
@@ -369,3 +369,13 @@ def test_bond_with_empty_ipv6_static_address(eth1_up):
+
+
+def test_create_vlan_over_a_bond(bond99_with_slave):
+    vlan_bond_slave_iface = 'eth2'
```
Please extract the slave name from `bond99_with_slave`.

EdDev commented on tests/integration/bond_test.py:
```diff
@@ -369,3 +369,13 @@ def test_bond_with_empty_ipv6_static_address(eth1_up):
+    with vlan_interface(
+        vlan_iface_name, vlan_id, vlan_bond_slave_iface
+    ) as desired_state:
+        assertlib.assert_state(desired_state)
```
You should also check that the bond state has not changed.

cathay4t commented on tests/integration/bond_test.py:
```diff
@@ -369,3 +370,17 @@ def test_bond_with_empty_ipv6_static_address(eth1_up):
+    vlan_id = 102
+    vlan_iface_name = '{}.{}'.format(vlan_bond_slave_iface, vlan_id)
+    with bond_interface(name='bond99', slaves=['eth2']) as bond_state:
+        assertlib.assert_state(bond_state)
```
Please move this line to the end of this function, so that we know the VLAN deletion does not impact bond.

EdDev commented on tests/integration/bond_test.py:
```diff
@@ -369,3 +370,17 @@ def test_bond_with_empty_ipv6_static_address(eth1_up):
+
+
+def test_create_vlan_over_a_bond_slave(bond99_with_slave):
+    vlan_bond_slave_iface = bond99_with_slave[Interface.KEY][0][
```
The name is confusing, how about: `bond_slave_ifname`.

Also, please make that long broken line readable by using intermediate variables:
```
bond_ifstate = bond99_with_slave[Interface.KEY][0]
bond_slave_ifname = bond_ifstate[Bond.CONFIG_SUBTREE][Bond.SLAVES][0]
```

EdDev commented on tests/integration/bond_test.py:
```diff
@@ -369,3 +370,17 @@ def test_bond_with_empty_ipv6_static_address(eth1_up):
+    ][Bond.SLAVES][0]
+    vlan_id = 102
+    vlan_iface_name = '{}.{}'.format(vlan_bond_slave_iface, vlan_id)
+    with bond_interface(name='bond99', slaves=['eth2']) as bond_state:
```
We do not need this, this is what bond99_with_slave is doing.

Nandan307 commented:
**Changelog:**

- Deleted ` bond_interface(name='bond99', slaves=['eth2'])` 
- assert_state now asserts `bond99_with_slave`
- Split the long `ifname` into smaller components as mentioned by Edy above

EdDev commented:
@cathay4t , seems like you comment has been addressed.
